DataManagementSystem/scripts/dirac-dms-directory-sync.py

```diff
@@ -0,0 +1,552 @@
+  dirs_delete.sort(key = lambda s: -s.count('/'))
+  #Create list of directories to create
+  dirs_create = list(from_dirs - to_dirs)
+  #Sort the list by depth of directory tree
+  dirs_create.sort(key = lambda s: s.count('/'))
+
+  #Flatten the list of pairs (filename, size) to list of filename
+  files_delete = [pair[0] for pair in list(to_files - from_files)]
+  files_create = [pair[0] for pair in list(from_files - to_files)]
+
+  create = {}
+  create["Directories"] = dirs_create
+  create["Files"] = files_create
+
+  delete = {}
+  delete["Directories"] = dirs_delete
+  delete["Files"] = files_delete
+
+  tree = {}
+  tree["Create"]=create
+  tree["Delete"]=delete
+
+  return S_OK(tree)
+
+
+def removeRemoteFiles(dm,lfns):
+  """
+  Remove file from the catalog
+  """
+  for lfnList in breakListIntoChunks( lfns, 100 ):
+    res = dm.removeFile( lfnList )
+    if not res['OK']:
+      return S_ERROR( "Failed to remove files:" + lfnList + res['Message'] )
+    else:
+      return S_OK()
+
+
+def uploadLocalFile(dm, lfn, localfile, storage):
+  """
+  Upload a local file to a storage element
+  """
+  res = dm.putAndRegister( lfn, localfile, storage, None )
+  if not res['OK']:
+    return S_ERROR( 'Error: failed to upload %s to %s' % ( lfn, storage ) )
+  else:
+    return S_OK( 'Successfully uploaded file to %s' % storage )
+
+def downloadRemoteFile(dm, lfn, destination):
+  """
+  Download a file from the system
+  """
+  res = dm.getFile( lfn, destination )
+  if not res['OK']:
+    return S_ERROR( 'Error: failed to download %s ' % lfn )
+  else:
+    return S_OK( 'Successfully uploaded file %s' % lfn )
+
+def removeStorageDirectoryFromSE( directory, storageElement ):
+  """
+  Delete directory on selected storage element
+  """
+
+  se = StorageElement( storageElement, False )
+  res = returnSingleResult( se.exists( directory ) )
+
+  if not res['OK']:
+    return S_ERROR( "Failed to obtain existence of directory" + res['Message'] )
+
+  exists = res['Value']
+  if not exists:
+    return S_OK( "The directory %s does not exist at %s " % ( directory, storageElement ) )
+
+  res = returnSingleResult( se.removeDirectory( directory, recursive = True ) )
+  if not res['OK']:
+    return S_ERROR( "Failed to remove storage directory" + res['Message'] )
+
+  return S_OK()
+
+def removeRemoteDirectory(fc,lfn):
+  """
+  Remove file from the catalog
+  """
+  storageElements = gConfig.getValue( 'Resources/StorageElementGroups/SE_Cleaning_List', [] )
+
+  for storageElement in sorted( storageElements ):
+    res = removeStorageDirectoryFromSE( lfn, storageElement )
+    if not res['OK']:
+      return S_ERROR( "Failed to clean storage directory at all SE:" + res['Message'] )
+  res = returnSingleResult( fc.removeDirectory( lfn, recursive = True ) )
+  if not res['OK']:
+    return S_ERROR( "Failed to clean storage directory at all SE:" + res['Message'] )
+
+  return S_OK("Successfully removed directory")
+
+
+def createRemoteDirectory(fc,newdir):
```

Comment:
This is usually not needed in DIRAC as the necessary directories are created automatically